### PR TITLE
Implement new radio buttons

### LINKED ISSUE #45761: Implement new radio buttons
[Radio Buttons](https://www.figma.com/file/Gpjs9vjhzUKF1GDbeG9JGE/Application-Design-System?type=design&node-id=8543-42749&mode=design&t=lfboeNMENiqaZygH-4): **_Size:_** lg, sm + **_Interaction states_**

### PR DESCRIPTION
**Pull request by bl-nero**
This PR adds a `RadioButton` component similar to `CheckboxInput`, as well as `FieldRadio` for composing radio buttons with labels and helper text, and then refactors `RadioGroup` to use it. It also tweaks the layout of small checkboxes slightly.

Screenshot 2024-08-23 at 19 04 40
Screenshot 2024-08-23 at 19 04 27
Screenshot 2024-08-23 at 19 14 40
Screenshot 2024-08-23 at 19 16 27

Deviations from the design:
- The focus state features a bold border to make it more pronounced. It's almost invisible otherwise.
- The unchecked radio border colors have been set to shades of grey in both themes (there's an inconsistency in Figma there).
- The focus state color has been altered to primary default to make it consistent with checkboxes.

Closes https://github.com/gravitational/teleport/issues/45761

### CODE REVIEW

File: web/packages/design/src/RadioButton/RadioButton.tsx [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+  // Give it some animation, but don't animate focus-related properties.
+  transition:
+    border-color 150ms,
+    background-color 150ms,
+    box-shadow 150ms;
+
+  // State-specific styles. Note: the "force" classes are required for
+  // Storybook, where we want to show all the states, even though we can't
+  // enforce them.
```
Comment (kiosion) [inline web/packages/design/src/RadioButton/RadioButton.tsx:168]: I thought triggering events/simulating user actions was possible thru Storybook? Could be misremembering though.

File: web/packages/design/src/RadioGroup/RadioGroup.tsx [hunk 2]
```diff
@@ -46,20 +49,30 @@ export function RadioGroup({
...
         return (
-          <Radio
-            onChange={onChange}
-            autoFocus={index === 0 && autoFocus}
+          <FieldRadio
             key={optionValue}
-            option={option}
             name={name}
+            label={optionLabel}
+            //helperText={...}
```
Comment (kiosion) [inline web/packages/design/src/RadioGroup/RadioGroup.tsx:68]:
```suggestion
```

**Review by kiosion — APPROVED** [review]
lgtm!

File: web/packages/design/src/RadioButton/RadioButton.tsx (hunk 1 above)
Comment (kiosion) [reply web/packages/design/src/RadioButton/RadioButton.tsx:168]: Nvm, they're within the same story, my bad 🙈

**Review by ryanclark — APPROVED** [review]

File: web/packages/design/src/RadioGroup/RadioGroup.tsx (hunk 2 above)
Comment (bl-nero) [reply web/packages/design/src/RadioGroup/RadioGroup.tsx:68]: Damn, you actually pointed out a small feature that I forgot to implement! :) I'm gonna do it in another PR, I guess.

**Pull request merged**